### PR TITLE
Remove SPMLibC from FileInfo

### PR DESCRIPTION
**Pull request by gmittert**

### CODE REVIEW

File: Sources/Basic/FileInfo.swift [hunk 1]
```diff
@@ -44,35 +37,24 @@ public struct FileInfo: Equatable, Codable {
...
-        let nanoseconds = buf.st_mtimespec.tv_nsec
-        #else
-        let seconds = buf.st_mtim.tv_sec
-        let nanoseconds = buf.st_mtim.tv_nsec
-        #endif
-
-        self.modTime = FileTimestamp(
-            seconds: UInt64(seconds), nanoseconds: UInt64(nanoseconds))
+    /// Kind of file system entity.
+    public let kind: Kind
```
Comment (aciidb0mb3r) [inline Sources/Basic/FileInfo.swift]: We can eliminate Kind and just store `FileAttributeType` directly, right?

File: Sources/Basic/FileInfo.swift [hunk 2]
```diff
@@ -44,35 +37,24 @@ public struct FileInfo: Equatable, Codable {
...
+
+    public init(_ attrs: [FileAttributeKey : Any]) {
+        self.device = attrs[.deviceIdentifier] as! UInt64
+        self.inode = attrs[.systemFileNumber] as! UInt64
+        self.posixPermissions = (attrs[.posixPermissions] as! NSNumber).int16Value
+        self.kind = Kind(type: attrs[.type] as! FileAttributeType)
+        self.size = attrs[.size] as! UInt64
+        self.modTime = attrs[.modificationDate] as! Date
```
Comment (aciidb0mb3r) [inline Sources/Basic/FileInfo.swift]: Is force-casting here safe? /cc @millenomi

File: Sources/Basic/FileInfo.swift [hunk 3]
```diff
@@ -8,30 +8,23 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
  */
 
-import SPMLibc
 import Foundation
 
 /// File system information for a particular file.
 public struct FileInfo: Equatable, Codable {
```
Comment (aciidb0mb3r) [inline Sources/Basic/FileInfo.swift]: `FileInfo` is stored in the manifest db so we need to bump the db schema: https://github.com/apple/swift-package-manager/blob/master/Sources/SPMLLBuild/llbuild.swift#L98

**Comment by jmittert** [comment]
Updated the db version and removed Kind.

**Comment by gmittert** [comment]
Updated to not force cast, instead making the fields optional.

**Comment by gmittert** [comment]
One Week Courtesy Ping

File: Sources/Basic/FileInfo.swift [hunk 4]
```diff
@@ -8,71 +8,37 @@
...
+    /// Kind of file system entity.
+    public let fileType: FileAttributeType?
+
+    public init(_ attrs: [FileAttributeKey : Any]) {
+        self.device = (attrs[.deviceIdentifier] as? NSNumber)?.uint64Value
+        self.inode = attrs[.systemFileNumber] as? UInt64
+        self.posixPermissions = (attrs[.posixPermissions] as? NSNumber)?.int16Value
+        self.fileType = attrs[.type] as? FileAttributeType
+        self.size = attrs[.size] as? UInt64
+        self.modTime = attrs[.modificationDate] as? Date
```
Comment (aciidb0mb3r) [inline Sources/Basic/FileInfo.swift]: I am a little concerned about this. Maybe put an assert so we at least get to know when these values are present?

File: Sources/Basic/FileInfo.swift [hunk 5]
```diff
@@ -8,71 +8,37 @@
...
-            case S_IFSOCK: self = .socket
-            default:
-                self = .unknown
-            }
-        }
-    }
-
-    /// The device number.
-    public let device: UInt64
+        /// The device number.
```
Comment (aciidb0mb3r) [inline Sources/Basic/FileInfo.swift]: Nit: This has an extra indentation.

File: Sources/Basic/FileInfo.swift (hunk 4 above)
Comment (gmittert) [reply Sources/Basic/FileInfo.swift]: I thought the idea was that the values may not be present?
```
Lily Vulcano [9:42 AM]
Force-casting isn’t safe; the file system may not return those values for one reason or another.
```
It makes sense to return nil in that case, doesn't it? Else I suppose I could make them all !s and have it crash if they're not there and we can figure out why they aren't there.

Comment (aciidb0mb3r) [reply Sources/Basic/FileInfo.swift]: Assertions are not triggered in release builds. I don't think a lot will go wrong if we don't have this info but it'll be nice to catch issues in debug builds which is usually what SwiftPM developers themselves use.

Comment (gmittert) [reply Sources/Basic/FileInfo.swift]: Alright, added in the asserts, and removed the optional typing of the fields.

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by gmittert** [comment]
@swift-ci please smoke test

**Pull request merged**